bidadjustment/apply_test.go

```diff
@@ -0,0 +1,447 @@
+			}
+			bidPrice, currencyAfterAdjustment := Apply(test.givenRuleToAdjustments, test.givenBidInfo, test.givenBidderName, bidCur, &reqInfo, test.givenBidType)
+			assert.Equal(t, test.expectedBidPrice, bidPrice, "Incorrect bid prices")
+			assert.Equal(t, test.expectedCurrency, currencyAfterAdjustment, "Incorrect currency")
+		})
+	}
+}
+
+type mockConversions struct {
+	mock.Mock
+}
+
+func (m mockConversions) GetRate(from string, to string) (float64, error) {
+	args := m.Called(from, to)
+	return args.Get(0).(float64), args.Error(1)
+}
+
+func (m mockConversions) GetRates() *map[string]map[string]float64 {
+	args := m.Called()
+	return args.Get(0).(*map[string]map[string]float64)
+}
+
+func TestApply(t *testing.T) {
+	var (
+		adjCur string = "EUR"
+		bidCur string = "USA"
+	)
+
+	testCases := []struct {
+		name             string
+		givenAdjustments []openrtb_ext.Adjustment
+		setMock          func(m *mock.Mock)
+		givenBidPrice    float64
+		expectedBidPrice float64
+		expectedCurrency string
+	}{
+		{
+			name:             "CpmAdjustment",
+			givenAdjustments: []openrtb_ext.Adjustment{{Type: AdjustmentTypeCpm, Value: 1.0, Currency: adjCur}},
+			givenBidPrice:    10.58687,
+			setMock:          func(m *mock.Mock) { m.On("GetRate", adjCur, bidCur).Return(2.5, nil) },
+			expectedBidPrice: 8.0869,
+			expectedCurrency: bidCur,
+		},
+		{
+			name:             "StaticAdjustment",
+			givenAdjustments: []openrtb_ext.Adjustment{{Type: AdjustmentTypeStatic, Value: 4.0, Currency: adjCur}},
+			givenBidPrice:    10.0,
+			setMock:          nil,
+			expectedBidPrice: 4.0,
+			expectedCurrency: adjCur,
+		},
+		{
+			name:             "MultiplierAdjustment",
+			givenAdjustments: []openrtb_ext.Adjustment{{Type: AdjustmentTypeMultiplier, Value: 3.0}},
+			givenBidPrice:    10.0,
+			setMock:          nil,
+			expectedBidPrice: 30.0,
+			expectedCurrency: bidCur,
+		},
+		{
+			name:             "ReturnOriginalPrice",
+			givenAdjustments: []openrtb_ext.Adjustment{{Type: AdjustmentTypeMultiplier, Value: -1.0}},
+			givenBidPrice:    10.0,
+			setMock:          nil,
+			expectedBidPrice: 10.0,
+			expectedCurrency: bidCur,
+		},
+		{
+			name:             "NilAdjustment",
+			givenAdjustments: nil,
+			givenBidPrice:    10.0,
+			setMock:          nil,
+			expectedBidPrice: 10.0,
+			expectedCurrency: bidCur,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			reqInfo := adapters.ExtraRequestInfo{}
+			if test.setMock != nil {
+				mockConversions := &mockConversions{}
+				test.setMock(&mockConversions.Mock)
+				reqInfo = adapters.NewExtraRequestInfo(mockConversions)
+			}
+
+			bidPrice, currencyAfterAdjustment := apply(test.givenAdjustments, test.givenBidPrice, bidCur, &reqInfo)
+			assert.Equal(t, test.expectedBidPrice, bidPrice, "Incorrect bid prices")
+			assert.Equal(t, test.expectedCurrency, currencyAfterAdjustment, "Incorrect currency")
+		})
+	}
+}
+
+func TestGet(t *testing.T) {
```

Comment:
Can you add one more test case here where none of the priority rules match the bid response?